src/utils/resources.py

```diff
@@ -1608,4 +1608,395 @@ def deprovision(self, context: Dict = {}):
+    - `$path_with_ruby` to be used in the systemd config (`Environment="PATH=__PATH_WITH_RUBY__"`)
+    - `$ruby_dir`, the directory containing the specific version of ruby, which may be used in the systemd config too (e.g. `ExecStart=__RUBY_DIR__/ruby foo bar`)
+
+    ### Example
+    ```toml
+    [resources.ruby]
+    version = "3.2"
+    ```
+
+    ### Properties
+    - `version`: The ruby version needed by the app
+
+    ### Provision/Update
+    - Fetch/update a copy of the rbenv tool as well as ruby-build, rbenv-aliases and xxenv-latest
+    - Compute the actual "latest" version for the requested version, e.g. `3.2` may corresponds to `3.2.1`
+    - This "actual version" is stored as `ruby_version` in the app settings
+    - Install (compile) Ruby (may take some time)
+    - Garbage-collect unused versions
+
+    ### Deprovision
+    - Delete the `ruby_version` setting
+    - Garbage-collect unused versions
+    """
+
+    type = "ruby"
+    priority = 100
+    version: str = ""
+
+    default_properties: Dict[str, Any] = {
+        "version": None,
+    }
+
+    RBENV_ROOT = "/opt/rbenv"
+
+    @property
+    def rbenv(self):
+        return f"{self.RBENV_ROOT}/bin/rbenv"
+
+    def installed_versions(self):
+
+        return check_output(
+            f"{self.rbenv} versions --bare --skip-aliases | grep -Ev '/'",
+            env={"RBENV_ROOT": self.RBENV_ROOT}
+        ).strip().split("\n")
+
+    def update_rbenv(self):
+
+        self._run_script(
+            "provision_or_update",
+            f"""
+            _ynh_git_clone "https://github.com/rbenv/rbenv" "{self.RBENV_ROOT}"
+            _ynh_git_clone "https://github.com/rbenv/ruby-build" "{self.RBENV_ROOT}/plugins/ruby-build"
+            _ynh_git_clone "https://github.com/tpope/rbenv-aliases" "{self.RBENV_ROOT}/plugins/rbenv-aliase"
+            _ynh_git_clone "https://github.com/momo-lab/xxenv-latest" "{self.RBENV_ROOT}/plugins/xxenv-latest"
+            mkdir -p "{self.RBENV_ROOT}/cache"
+            mkdir -p "{self.RBENV_ROOT}/shims"
+        """
+        )
+
+    def provision_or_update(self, context: Dict = {}):
+
+        for package in ["gcc", "make", "libjemalloc-dev", "libffi-dev", "libyaml-dev", "zlib1g-dev"]:
+            if os.system(f'dpkg --list | grep -q "^ii  {package}"') != 0:
+                raise YunohostValidationError(f"{package} is required to install Ruby")
+
+        self.update_rbenv()
+
+        ruby_version = check_output(
+            f"{self.rbenv} latest --print '{self.version}'",
+            env={"RBENV_ROOT": self.RBENV_ROOT},
+        )
+        self.set_setting("ruby_version", ruby_version)
+        logger.info(f"Building Ruby {ruby_version}, this may take some time...")
+        self._run_script(
+            "provision_or_update",
+            f"""
+            #export RBENV_ROOT='{self.RBENV_ROOT}'
+            export RUBY_CONFIGURE_OPTS='--disable-install-doc --with-jemalloc'
+            export MAKE_OPTS='-j2'
```

Comment:
Does it takes time to compile? Should we use number of cores instead?

Reply:
Yeah I was wondering too about where this `2` comes from, it's purely a copypasta of the options in the current helpers ... I dunno what we want to do exactly, maybe it is only 2 to not overload the server ... but maybe using half of the available core could be smarter ?

Anyway it seem to take time to compile indeed, on my test machine (thinkpad) it did took like 10-20 minutes I think ? And it does take significant time on the CI too because it does compile Ruby :/ ... (I don't understand why there can't be pre-compiled versions like Debian ships ...)

Reply:
```suggestion
            # Use half of available CPUs for compiling
            # The trick with 1 + ...  -1 is to prevent having '0' when there's only one CPU available.
            NB_CPU_TO_USE="$(( 1 + ($(grep -c ^processor /proc/cpuinfo) - 1) / 2 ))"
            export MAKE_OPTS="-j$(NB_CPU_TO_USE)"
```